validphys2/src/validphys/theorycovariance/construction.py

```diff
@@ -199,6 +225,234 @@ def covmat_9pt(name1, name2, deltas1, deltas2):
+            # Locate position of the experiment
+            size = process_info.sizes[exp]
+            start_proc_by_exp[exp] = running_index_tot
+            running_index_tot += size
+            running_index_proc += size
+
+            # Compute shifts only for a subset of processes
+            if proc in filtered_proc and exp not in excluded_exp:
+                #central = process_info.preds[proc][1][start_proc_by_exp[exp] : size] # Probably this is deprecated
+                x = process_info.data[proc].T[0][running_index_proc - size : running_index_proc]
+                Q2 = process_info.data[proc].T[1][running_index_proc - size : running_index_proc] 
+                y = process_info.data[proc].T[2][running_index_proc - size : running_index_proc]
+
+                # TO UNDERSTAND: how can I retrieve the observable from the name of the experiment
+                if "SIGMA" in exp:
+                    N_2, N_L = compute_normalisation_by_experiment(exp, x=x, y=y, Q2=Q2)
+
+                elif "F2":
+                    N_2 = np.ones(shape=x.shape)
+                    N_L = np.zeros(shape=x.shape)
+
+                else:
+                    raise ValueError(f"The normalisation for the observable ... is not known.")
+
+                if ht_pt_prescription == 5:
+                    deltas["(+,0)"] += [N_2 * ht_coeff_1 * H_2(x)]
+                    deltas["(0,+)"] += [N_L* ht_coeff_2 * H_L(x)]
+                else:
+                    raise ValueError(
+                        f"The pt prescription for the HT theory covmat is not supported."
+                    )
+
+    # Construct theory covmat
+    covmats = defaultdict(list)
+    for proc1 in filtered_proc:
+        for proc2 in filtered_proc:
+            for i, exp1 in enumerate(process_info.namelist[proc1]):
+                for j, exp2 in enumerate(process_info.namelist[proc2]):
+                    if ht_pt_prescription == 5:
+                        s = np.outer(deltas["(+,0)"][i], deltas["(+,0)"][j]) + \
+                            np.outer(deltas["(0,+)"][i], deltas["(0,+)"][j])
+                    start_locs = (start_proc_by_exp[exp1], start_proc_by_exp[exp2])
+                    covmats[start_locs] = s
+    return covmats
+
+
+def compute_normalisation_by_experiment(experiment_name, **kwargs):
+    N_2 = np.zeros(shape=kwargs.get('y').shape)
+    N_L = np.zeros(shape=kwargs.get('y').shape)
+
+    if "HERA_NC" in experiment_name or "HERA_CC" in experiment_name:
+        y = kwargs.get('y')
+        yp = 1 + np.power(1 - y, 2)
+        yL = np.power(y, 2)
+
+        if "HERA_NC" in experiment_name:
+            N_2 = 1
+            N_L = - yL / yp
+
+        elif "HERA_CC" in experiment_name:
+            N_2 = 1 / 4 * yp
+            N_L = - N_2 * yL / yp
+
+    if "CHORUS_CC" in experiment_name:
+        x = kwargs.get('xB')
+        y = kwargs.get('y')
+        Q2 = kwargs.get('Q2')
+        yL = np.power(y, 2)
+        Gf = 1
+        Mh = 1
+        MW2 = 1
```

Comment:
- Gf is Fermi's constant, the value can be copied from the theorycard 
- Mh is the proton mass, also in the theory card
- MW2 is the W boson mass squard, also in the theory card